infra/Pos/Slotting/MemState/Class.hs

```diff
@@ -8,24 +8,52 @@ import           Control.Monad.Trans (MonadTrans)
 import           Universum
 
 import           Pos.Core.Types      (EpochIndex, Timestamp)
-import           Pos.Slotting.Types  (SlottingData)
+import           Pos.Slotting.Types  (EpochSlottingData)
+
 
 -- | 'MonadSlotsData' provides access to data necessary for slotting to work.
+-- Both _current_ and _next_ epoch @SlottingData@ need to be present since it's impossible
+-- to create @SlottingData@ without them.
 class Monad m => MonadSlotsData m where
 
-    getSystemStart :: m Timestamp
+    getSystemStartM :: m Timestamp
+
+    getAllEpochIndicesM :: m [EpochIndex]
+
+    getCurrentEpochIndexM :: m EpochIndex
+
+    getNextEpochIndexM :: m EpochIndex
 
-    getSlottingData :: m SlottingData
+    getCurrentEpochSlottingDataM :: m EpochSlottingData
 
-    waitPenultEpochEquals :: EpochIndex -> m ()
+    getNextEpochSlottingDataM :: m EpochSlottingData
 
-    putSlottingData :: SlottingData -> m ()
+    getEpochSlottingDataM :: EpochIndex -> m (Maybe EpochSlottingData)
+
+    putEpochSlottingDataM :: EpochIndex -> EpochSlottingData -> m ()
+
+    waitCurrentEpochEqualsM :: EpochIndex -> m ()
```

Comment:
Ok, so I have nothing logic-related to say except for things already mentioned by pva/gromak. 
Styling comments:
1. Why do you adopt `methodM` pattern here? What's the point?
2. It'd be good to provide at least short documentation to methods here since you've figured out how do they work.
3. Spaces between method declarations are ugly and don't make sense.